Update DG and edit UG (#130)
AY1920S1-CS2103T-T13-4/main: docs/DeveloperGuide.adoc

[TIP]
The `.puml` files used to create diagrams in this document can be found in the link:{repoURL}/docs/diagrams/[diagrams] folder.

`Main` has two classes called link:{repoURL}/src/main/java/seedu/mark/Main.java[`Main`] and link:{repoURL}/src/main/java/seedu/address/MainApp.java[`MainApp`]. It is responsible for,

* Initializes the components in the correct sequence, and connects them up with each other.
* Shuts down the components and invokes cleanup method where necessary.

<<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.

> Like `LogsCenter` plays an important role at the architecture level. `LogsCenter` is used to write log messages to the App's log file.

Reviewer: ```suggestion
For example, `LogsCenter` plays an important role at the architecture level as it is used by many classes to write log messages to the App's log file.
```

> 

Reviewer: ![image](https://user-images.githubusercontent.com/42093424/67396588-710f1400-f5da-11e9-8c44-e148cbca1170.png)
Should we update this as well?

Author: Maybe it means for future release?

[TIP]
The `.puml` files used to create diagrams in this document can be found in the link:{repoURL}/docs/diagrams/[diagrams] folder.

> `Main` has two classes called link:{repoURL}/src/main/java/seedu/address/Main.java[`Main`] and link:{repoURL}/src/main/java/seedu/address/MainApp.java[`MainApp`]. It is responsible for,

Reviewer: ```suggestion
`Main` has two classes called link:{repoURL}/src/main/java/seedu/mark/Main.java[`Main`] and link:{repoURL}/src/main/java/seedu/address/MainApp.java[`MainApp`]. It is responsible for,
```

* Initializes the components in the correct sequence, and connects them up with each other.
* Shuts down the components and invokes cleanup method where necessary.

<<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.
`LogsCenter` plays an important role at the architecture level. Here is how it works:

> * Uses `LogsCenter` to write log messages to the App's log file.

Reviewer 2: ```suggestion
* `LogsCenter` is used by many classes to write log messages to the App's log file.
```

because now i'm not sure who uses it if it starts with "Uses"

Author: Okay. But I think this description still need to change because other descriptions describe how the high-level design works, but this one seems like how the inner part works.

> The rest of the App consists of four components.

Reviewer: ```suggestion
The rest of the App consists of the following four components:
```

> * Uses `LogsCenter` to write log messages to the App's log file.

Reviewer: ![image](https://user-images.githubusercontent.com/42093424/67396792-c9461600-f5da-11e9-97c5-5dd7a83d4ab0.png)

I feel it is not necessary to have this as a single bullet here since usually there should be several bullets.

Author: Okay

// end::dataencryption[]

// tag::reminder[]
=== Reminder Feature
==== Implementation

> A bookmark can have a reminder, and a reminder can use to open the bookmark.

Reviewer 2: ```suggestion
A bookmark can have a reminder, and a reminder can be used to open the bookmark.
```

This mechanism is facilitated by `Reminder` and `ReminderAssociation`.

> `Reminder` containing a reminding time, the `Url` of the `Bookmark` and a `Note`.

Reviewer 2: ```suggestion
`Reminder` contains a reminding time, the `Url` of the `Bookmark` and a `Note`.
```

`ReminderAssociation` represents the association between `Reminder` and `Bookmark`, containing
 the relation from `Reminder` to `Bookmark` and `Reminder` to `Bookmark`.

.Structure of the Reminder
image::ReminderAssociationDiagram.png[]

> ==== Design Consideration

Reviewer 2: ```suggestion
==== Design Considerations
```

===== Aspect: How to represent the relationship between reminder and bookmark.

> * **Alternative 1 (current choice):** Use two hashmap to store both the relationship from reminder to bookmark and bookmark to reminder.

Reviewer 2: ```suggestion
* **Alternative 1 (current choice):** Use two hashmaps to store both the relationship from reminder to bookmark and bookmark to reminder.
```

** Pros: More OOP and can easily find the relationship.

> ** Cons: More complex when doing any operation to a reminder. Need to check two hashmap.

Reviewer 2: ```suggestion
** Cons: More complex when doing any operation to a reminder. Need to check two hashmaps.
```

* **Alternative 2:** Reminder is a field of bookmark.
** Pros: Easier to implement.

> ** Cons: Reminder is a field of bookmark, so using a reminder to open the bookmark violate the relationship.

Reviewer 2: violates what relationship ah?

Author: It like Bookmark --> Reminder, but now using reminder to open bookmark will be Reminder --> Bookmark. I don't how to describe it precisely.

Author: ```suggestion
//TODO: Change this description
** Cons: Reminder is a field of bookmark, so using a reminder to open the bookmark violate the relationship.
```

// end::dataencryption[]

// tag::reminder[]
=== Reminder Feature
==== Implementation

> A bookmark can have a reminder, and a reminder can be used to open the bookmark.

Reviewer: ```suggestion
A bookmark can have a reminder, and a reminder can be used to open the bookmark at a specific time.
```

Author: I think open at a specific time is a "good to have" feature, currently it will just show on the dashboard at the order of time.

This mechanism is facilitated by `Reminder` and `ReminderAssociation`.

`Reminder` contains a reminding time, the `Url` of the `Bookmark` and a `Note`.

`ReminderAssociation` represents the association between `Reminder` and `Bookmark`, containing
 the relation from `Reminder` to `Bookmark` and `Reminder` to `Bookmark`.

.Structure of the Reminder
image::ReminderAssociationDiagram.png[]

==== Design Considerations

> ===== Aspect: How to represent the relationship between reminder and bookmark.

Reviewer: ```suggestion
===== Aspect: How to represent the relationship between reminder and bookmark
```
